library/src/main/java/com/novoda/downloadmanager/FileCallbackThrottleByTime.java

```diff
@@ -19,7 +19,7 @@ public void setCallback(final DownloadBatchStatusCallback callback) {
     @Override
     public void update(DownloadBatchStatus downloadBatchStatus) {
         if (callback == null) {
-            return;
+            throw new IllegalStateException("A DownloadBatchStatusCallback must be set before an update is called.");
```

Comment:
I'm a little worried about introducing this change but it is a valid change that we should make.

Reply:
same as before